### ISSUE TITLE
Error on checking of CUDA files

### ISSUE THREAD

**Issue by derpda** [vim-syntastic/syntastic#2174]
# Version information
## Output of :version

```
VIM - Vi IMproved 7.4 (2013 Aug 10)
Included patches: 1-207, 209-326
Compiled by 'http://www.opensuse.org/'
Huge version without GUI.  Features included (+) or not (-):
+acl             +cmdline_hist    -dnd             -footer          +lispindent      -mouse_jsbterm   +perl            +scrollbind      -tcl             +viminfo         -xterm_save
+arabic          +cmdline_info    -ebcdic          +fork()          +listcmds        +mouse_netterm   +persistent_undo +signs           +terminfo        +vreplace        -xpm
+autocmd         +comments        +emacs_tags      +gettext         +localmap        +mouse_sgr       +postscript      +smartindent     +termresponse    +wildignore
-balloon_eval    +conceal         +eval            -hangul_input    +lua/dyn         -mouse_sysmouse  +printer         +sniff           +textobjects     +wildmenu
-browse          +cryptv          +ex_extra        +iconv           +menu            +mouse_urxvt     +profile         +startuptime     +title           +windows
++builtin_terms  +cscope          +extra_search    +insert_expand   +mksession       +mouse_xterm     +python/dyn      +statusline      -toolbar         +writebackup
+byte_offset     +cursorbind      +farsi           +jumplist        +modify_fname    +multi_byte      +python3/dyn     -sun_workshop    +user_commands   -X11
+cindent         +cursorshape     +file_in_path    +keymap          +mouse           +multi_lang      +quickfix        +syntax          +vertsplit       -xfontset
-clientserver    +dialog_con      +find_in_path    +langmap         -mouseshape      -mzscheme        +reltime         +tag_binary      +virtualedit     -xim
-clipboard       +diff            +float           +libcall         +mouse_dec       +netbeans_intg   +rightleft       +tag_old_static  +visual          -xsmp
+cmdline_compl   +digraphs        +folding         +linebreak       -mouse_gpm       +path_extra      +ruby/dyn        -tag_any_white   +visualextra     -xterm_clipboard
   system vimrc file: "/etc/vimrc"
     user vimrc file: "$HOME/.vimrc"
 2nd user vimrc file: "~/.vim/vimrc"
      user exrc file: "$HOME/.exrc"
  fall-back for $VIM: "/etc"
 f-b for $VIMRUNTIME: "/usr/share/vim/current"
Compilation: gcc -c -I. -Iproto -DHAVE_CONFIG_H     -fmessage-length=0 -grecord-gcc-switches -fstack-protector -O2 -Wall -D_FORTIFY_SOURCE=1 -funwind-tables -fasynchronous-unwind-tables -g -W
all -pipe -fno-strict-aliasing
Linking: gcc   -L. -fstack-protector -rdynamic -Wl,-export-dynamic -Wl,-E -Wl,-rpath,/usr/lib/perl5/5.18.2/x86_64-linux-thread-multi/CORE   -L/usr/local/lib -Wl,--as-needed -o vim        -lm
-lnsl    -ltinfo -lacl -lattr -ldl   -Wl,-E -Wl,-rpath,/usr/lib/perl5/5.18.2/x86_64-linux-thread-multi/CORE  -L/usr/local/lib64 -fstack-protector  -L/usr/lib/perl5/5.18.2/x86_64-linux-thread-
multi/CORE -lperl -lm -ldl -lcrypt -lpthread
```

## Output of :SyntasticInfor
```
Syntastic version: 3.9.0 (Vim 704, Linux)
Info for filetype: cuda
Global mode: active
Filetype cuda is active
The current file will be checked automatically
Available checker: nvcc
Currently enabled checker: nvcc
```
# Steps to reprocude
No change in settings for the CUDA checkker in .vimrc.
When checking a CUDA file (ie on opening it if `g:syntastic_check_on_open = 1`), I get the following error:

```
Error detected while processing function <SNR>28_BufEnterHook..<SNR>28_UpdateErrors..<SNR>28_CacheErrors..18..17..SyntaxCheckers_cuda_nvcc_GetLocList:
line   22:
E116: Invalid arguments for function syntastic#c#ReadConfig(syntastic#util#bufVar(buf, 'cuda_config_file'), 'args_after': '-Xcompiler -fsyntax-only', 'tail_after': syntastic#c#NullOutput() })

E116: Invalid arguments for function extend
Error detected while processing function <SNR>28_BufEnterHook..<SNR>28_UpdateErrors..<SNR>28_CacheErrors:
line   39:
E170: Missing :endfor
Error detected while processing function <SNR>28_BufEnterHook..<SNR>28_UpdateErrors:
line   15:
E171: Missing :endif
Error detected while processing function <SNR>28_BufEnterHook:
line    9:
E171: Missing :endif
```

# Possible solution

I think the following line is to blame.
https://github.com/vim-syntastic/syntastic/blob/e53722e61d68aae278d75191038c6fda3777f3d5/syntax_checkers/cuda/nvcc.vim#L40

A bracket seems to be missing.

**Comment by lcd047** [comment]
Indeed, thank you.  Fixed in d17b98cf.

**Comment by derpda** [comment]
My bad. Ignore the pull request I just posted!
